Split DRM output instance from the DRM platform EGL interface

petedavis opened the pull request:
## What does the pull request do?
Step towards dual screen displays is to split the platform EGL interface from the DrmOutput instances which are used for surfaces for each display

## What is the current behavior?
DrmOutput is the platform OpenGl provider, and single surface instance.

## What is the updated/expected behavior with this PR?
Now able to create multiple DrmOutput instances for multiple connectors.

## Checklist

- [ ] Added unit tests (if possible)?
- [x] Added XML documentation to any related classes?
- [x] Consider submitting a PR to https://github.com/AvaloniaUI/Avaloniaui.net with user documentation

## Breaking changes
No breaking changes. ControlCatalog.NetCore still runs with the `--drm` command line args, and no changes required.

petedavis committed:
Split DRM output instance from the DRM platform EGL interface

Step towards dual screen displays is to split the platform EGL interface from the DrmOutput instances which are used for surfaces for each display

kekekeks commented on src/Linux/Avalonia.LinuxFramebuffer/Output/DrmBindings.cs:
```diff
@@ -149,8 +150,24 @@ public DrmCard(string path = null)
+        /// <summary>
+        /// Pointer to the Graphics Buffer Management (GBM) Device for the DrmCard.  
+        /// </summary>
+        internal IntPtr GbmDevice
```
GBM device handle should be either created in constructor or passed to constructor. In this case it should probably be a separate class with its own resource management.

petedavis replied:
This has now been wrapped in a GbmDevice class. I have still had to expose a handle property though. This can only really be removed if the GbmDevice is responsible for creating the Egl Platform Interface instance from it.

kekekeks commented on src/Linux/Avalonia.LinuxFramebuffer/Output/DrmPlatform.cs:
```diff
@@ -0,0 +1,66 @@
+        {
+        }
+
+        public DrmPlatform(DrmCard card, double scaling = 1.0)
```
Since you are adding multiple output support, scaling should be configured for each output separately

petedavis replied:
Done.

kekekeks commented on src/Linux/Avalonia.LinuxFramebuffer/LinuxFramebufferPlatform.cs:
```diff
@@ -13,49 +14,61 @@
+
+        private static string OutputPlatformName(IOutputBackend outputBackend)
+        {
+            var name = outputBackend.GetType().Name.ToLowerInvariant();
```
That should be a property on IOutputBackend

petedavis replied:
Done

petedavis committed:
Create name property for the output backend type

petedavis committed:
Encapsulate GBM device in its own class.

petedavis committed:
Allow for scaling per DRM display output

petedavis committed:
Ensure DRM platform OpenGL interface is registered.

petedavis committed:
Fix segmentation fault when closing app with two displays

petedavis committed:
Merge branch 'master' into split-drm-surfaces-from-egl-platorm

petedavis committed:
Enable custom use of custom lifetimes with the LinuxFramebufferPlatform

Ensures internally that the platform is initialised, with the caller able to create their own lifetime for the platform.

grokys commented:
@petedavis @kekekeks what's the status on this?
